helm/templates/gateway/_gateway-deployment-probes.tpl

```diff
@@ -0,0 +1,107 @@
+    {{- $gatewayServiceCoreScheme }}
+{{- end }}
+
+{{- define "gateway.computeLivenessProbe" -}}
+    {{- $httpHeadersProbes := (include "gateway.httpHeadersProbes" . | fromYaml) -}}
+    {{- $gatewayServiceCoreScheme := (include "gateway.gatewayServiceCoreScheme" .) -}}
+
+    {{- $defaultLivenessProbe := dict
+        "initialDelaySeconds" 30
+        "periodSeconds" 30
+        "timeoutSeconds" 3
+        "successThreshold" 1
+        "failureThreshold" 3
+        "httpGet" (merge (dict
+          "path" "/_node/health?probes=http-server"
+          "scheme" $gatewayServiceCoreScheme
+          "port" (.Values.gateway.services.core.http.port | default 18082)
+        ) $httpHeadersProbes)
+    }}
+
+    {{- $computeLivenessProbe := (mergeOverwrite $defaultLivenessProbe .Values.gateway.livenessProbe) -}}
+    {{- if or
+            (hasKey $computeLivenessProbe "tcpSocket")
+            (hasKey $computeLivenessProbe "exec")
+    }}
+      {{- $_ := unset $computeLivenessProbe "httpGet" }}
+    {{- end }}
+
+    {{- $computeLivenessProbe | toYaml }}
+{{- end }}
+
+{{- define "gateway.computeReadinessProbe" -}}
+    {{- $httpHeadersProbes := (include "gateway.httpHeadersProbes" . | fromYaml) -}}
+    {{- $gatewayServiceCoreScheme := (include "gateway.gatewayServiceCoreScheme" .) -}}
+    {{- $httpGetPath := (ternary "/_node/health?probes=security-domain-sync" "/_node/health?probes=http-server" (eq .Values.gateway.readinessProbe.domainSync true)) -}}
+
+    {{- $defaultReadinessProbe := dict
+        "initialDelaySeconds" 30
+        "periodSeconds" 30
+        "timeoutSeconds" 3
+        "successThreshold" 1
+        "failureThreshold" 3
+        "httpGet" (merge (dict
+          "path" $httpGetPath
+          "scheme" $gatewayServiceCoreScheme
+          "port" (.Values.gateway.services.core.http.port | default 18082)
+        ) $httpHeadersProbes)
+    }}
+
+    {{- $computeReadinessProbe := (mergeOverwrite $defaultReadinessProbe .Values.gateway.readinessProbe) -}}
+    {{- if or
+            (hasKey $computeReadinessProbe "tcpSocket")
+            (hasKey $computeReadinessProbe "exec")
+    }}
+      {{- $_ := unset $computeReadinessProbe "httpGet" }}
+    {{- end }}
+
+    {{- $_ := unset $computeReadinessProbe "domainSync" }}
+
+    {{- $computeReadinessProbe | toYaml }}
+{{- end }}
+
+{{- define "gateway.computeStartupProbe" -}}
+    {{- $httpHeadersProbes := (include "gateway.httpHeadersProbes" . | fromYaml) -}}
+    {{- $gatewayServiceCoreScheme := (include "gateway.gatewayServiceCoreScheme" .) -}}
+
+    {{- $defaultStartupProbe := dict
+        "initialDelaySeconds" 10
+        "periodSeconds" 10
+        "timeoutSeconds" 1
+        "successThreshold" 1
+        "failureThreshold" 29
+        "httpGet" (merge (dict
+          "path" "/_node/health?probes=http-server,security-domain-sync"
+          "scheme" $gatewayServiceCoreScheme
+          "port" (.Values.gateway.services.core.http.port | default 18082)
```

Comment:
**issue**: does this default port is the right one ? 18082 is usually the one for APIM. AM ius using 18092 or 18093 according to the service (cf removed default values from the values.yaml)